catalog/transformer/catalog-transformer-streaming-lib/src/main/java/org/codice/ddf/transformer/xml/streaming/lib/SaxEventToXmlElementConverter.java

```diff
@@ -213,8 +219,21 @@ public SaxEventToXmlElementConverter reset() {
      * @param uri    the namespace uri that is passed in by {@link SaxEventHandler}
      */
     public void addNamespace(String prefix, String uri) throws XMLStreamException {
-        out.setPrefix(prefix, uri);
         namespaces.put(prefix, uri);
+        namespaceStack.push(new ImmutablePair<>(prefix, uri));
+    }
+
+    public void removeNamespace(String prefix) {
+        Iterator<ImmutablePair<String, String>> iter = namespaceStack.iterator();
+        while (iter.hasNext()) {
+            Pair pair = iter.next();
+            if (pair.getKey()
+                    .equals(prefix)) {
+                iter.remove();
+                break;
```

Comment:
That's actually what this PR fixes. This is supposed to just pop off the first one (or most recent) so that scope is preserved. You can see in the unit test, when `removeNamespace` is called, there are/would be two of "ns1" in the `namespaceStack` but only one is removed, otherwise the rest of the code would fail.